darwin/PlatformHelpers.c

```diff
@@ -0,0 +1,124 @@
+/*
+htop - DarwinProcessList.c
+(C) 2014 Hisham H. Muhammad, 2021 Alexander Momchilov
+Released under the GNU GPLv2, see the COPYING file
```

Comment:
Incorrect copyright line, cf. #808 …

Reply:
IDK how that works, should I be putting my name on the file I edited?

and the second line should be:

```
Released under the GNU GPLv2+, see the COPYING file
```

Right?

Reply:
As this file has just been created it doesn't make sense to copyright it to 2014. Have a look at [the styleguide](https://github.com/htop-dev/htop/blob/main/docs/styleguide.md#file-content-structure). The second part of that line with your name is okay.

Reply:
> As this file has just been created it doesn't make sense to copyright it to 2014.

I didn't know how this works. I figured that because the content already existed from 2014, and was just being moved, that it should remain "2014".

Reply:
Well, in that case you might have to look through the history/blame of the original file to identify who touched that particular code. This is likely to involve Hisham, but may also include others. There are likely some changes from last year that warrant the inclusion of the htop dev team part for 2020-2021; which was mostly why I stumbled over this line in the first place (apart from the license mention). I'll also have a (closer) look into this later too.

Reply:
Okay, going through the codebase the copyright line for this new file probably should read like this:
```diff
diff --git a/darwin/PlatformHelpers.c b/darwin/PlatformHelpers.c
index 1ef42b8f..c549f770 100644
--- a/darwin/PlatformHelpers.c
+++ b/darwin/PlatformHelpers.c
@@ -1,6 +1,6 @@
 /*
 htop - DarwinProcessList.c
-(C) 2021 Hisham H. Muhammad, Alexander Momchilov
+(C) 2018 Pierre Malhaire, 2020-2021 htop dev team, 2021 Alexander Momchilov
 Released under the GNU GPLv2+, see the COPYING file
 in the source distribution for its full text.
 */
@@ -91,7 +91,7 @@ double Platform_calculateNanosecondsPerMachTick() {
    // If the API is unavailable assume we get our timebase in nanoseconds.
 #ifndef HAVE_MACH_TIMEBASE_INFO
    return 1.0;
-#endif
+#else
 
    mach_timebase_info_data_t info;
 
@@ -122,4 +122,5 @@ double Platform_calculateNanosecondsPerMachTick() {
    }
 
    return (double)info.numer / (double)info.denom;
+#endif
 }
```

Short breakdown of that copyright line:
* 23f96048c74a5a958dfb20b37eba9cde35a4e97d introduced the GetKernelVersion workaround (by Pierre Malhaire)
* f32f0188cd140f965c5e02f7fb875116522c4b41 introduced the timebase fix for #368 (by me, attributed to the htop dev team)
* several commits by you this year

Also that patch introduces a small fix in `Platform_calculateNanosecondsPerMachTick` to avoid dead code after the initial `return` statement.

Reply:
I've applied those changes, and also added the same copyright line to the matching header

BTW, have you heard of the "suggestions" feature on GitHub?

https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request

Reply:
Sure, using it regularly. But with this discussion it would have affected both lines (not just the actual copyright line). Also the feature hase the drawback that it only allows suggestions for lines surrounding lines changed in a PR. Thus for the #ifdef/#endif change I had in my patch this would have been somewhat strange. btw. apart from this maintainers of a repository you open a PR in can (unless you deselect one checkbox) edit the patches in the PR'ing repo by updating the branch that should be merged. I decided against these options in this case to have someone else check the copyright status too (I got ACK by @fasterit off-band).